internal/datacoord/meta.go

```diff
@@ -218,11 +238,17 @@ func (m *meta) GetAllSegment(segID UniqueID) *SegmentInfo {
 
 // SetState setting segment with provided ID state
 func (m *meta) SetState(segmentID UniqueID, targetState commonpb.SegmentState) error {
+	log.Info("meta update: setting segment state",
+		zap.Int64("segment ID", segmentID),
+		zap.Any("target state", targetState))
 	m.Lock()
 	defer m.Unlock()
 	curSegInfo := m.segments.GetSegment(segmentID)
 	if curSegInfo == nil {
-		// TODO: Should return error instead.
+		log.Info("meta update: setting segment state - segment not found",
+			zap.Int64("segment ID", segmentID),
+			zap.Any("target state", targetState))
+		// TODO: Should probably return error instead.
 		return nil
```

Comment:
Any idea why no error is returned here?

Reply:
Could be a bug, will investigate and see if it is safe to return an error later.